soh/src/code/z_parameter.c

```diff
@@ -1851,9 +1851,12 @@ u8 Item_Give(PlayState* play, u8 item) {
             }
             
         } else if (item == ITEM_SWORD_MASTER) {
+            //this forces the sword to the Master Sword, despite the fact I don't think this is needed. Breaks DList Rework, but messes with rando / timeless equipment?
+            /*
             gSaveContext.equips.buttonItems[0] = ITEM_SWORD_MASTER;
             gSaveContext.equips.equipment &= 0xFFF0;
             gSaveContext.equips.equipment |= 0x0002;
+            */
```

Comment:
it seems these lines have been around since the repo went public https://github.com/HarbourMasters/Shipwright/blame/develop/soh/src/code/z_parameter.c#L1854

Reply:
I'd assume so. My guess is that it's (redundant) vanilla behavior. Potentially a holdover from an earlier build of the game when it WAS needed.

Reply:
Without a proper decomp of a non-debug rom, we cannot make assumptions on what is a "holdover". Removing this is changing authentic behavior of the game.

In this case, this block is responsible for setting the master sword as your active sword when picking it up during the Ganon fight. We do not want to remove this.

Reply:
The problem is that this **also** forces the sword to the Master Sword as Young Link draws the master sword, forcing his sheath to the MS sheath.

Reply:
At the very least, I've uncommented the code to make sure the ganon fight works

Reply:
Should be fixed now.